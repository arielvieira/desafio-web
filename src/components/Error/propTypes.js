import PropTypes from 'prop-types';

const ErrorPropTypes = {
  error: PropTypes.string,
};

export default ErrorPropTypes;
