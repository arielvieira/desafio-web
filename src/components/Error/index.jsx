import React from 'react';

import propTypes from './propTypes';

const Error = ({ error }) => (
  <div className="error">
    {error}
  </div>
);

Error.defaultProps = {
  error: '',
};

Error.propTypes = {
  error: propTypes.error,
};

export default Error;
