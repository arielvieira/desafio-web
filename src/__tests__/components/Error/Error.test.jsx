import React from 'react';
import { shallow } from 'enzyme';

import Error from '../../../components/Error';

describe('Error', () => {
  it('should render Error correctly', () => {
    const wrapper = shallow(<Error error="API rate limit exceeded for 191.54.128.217." />);
    expect(wrapper).toMatchSnapshot();
  });
});
