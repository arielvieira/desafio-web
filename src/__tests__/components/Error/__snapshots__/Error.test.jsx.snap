// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render Error correctly 1`] = `
<div
  className="error"
>
  API rate limit exceeded for 191.54.128.217.
</div>
`;
